cmd/kops/create_cluster.go

```diff
@@ -497,6 +502,13 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 		}
 		zoneToSubnetMap[region] = subnet
 	} else {
+		var zoneToSubnetProviderID map[string]string
+		if len(c.Zones) > 0 && len(c.SubnetIDs) > 0 {
+			zoneToSubnetProviderID, err = getZoneToSubnetProviderID(c.VPCID, c.Zones[0][:len(c.Zones[0])-1], c.SubnetIDs)
```

Comment:
I think we should check that `api.CloudProviderID(cluster.Spec.CloudProvider) == api.CloudProviderAWS`, until we support this flag on other clouds